app/controllers/spree/amazon_controller.rb

```diff
@@ -114,14 +122,18 @@ def update_payment_amount!
   end
 
   def complete_amazon_order!
-    amazon_order.set_order_reference_details(
-      current_order.total,
-      seller_order_id: current_order.number,
-      store_name: current_order.store.name,
-    )
-    if amazon_order.confirm.success
+    unless @order.amazon_transaction.retry
+      amazon_order.set_order_reference_details(
+        current_order.total,
+        seller_order_id: current_order.number,
+        store_name: current_order.store.name,
+      )
+    end
+    
+    confirm_response = amazon_order.confirm
+    if confirm_response.success
       amazon_order.fetch
-
+      
```

Comment:
trim extra space. 

I use the [Traling Spaces](https://github.com/SublimeText/TrailingSpaces) extension for sublime to automatically get rid of those.